internal/adapters/entrypoints/rest/handlers/set_pegout_config.go

```diff
@@ -25,11 +29,15 @@ func NewSetPegoutConfigHandler(useCase *liquidity_provider.SetPegoutConfigUseCas
 
 		err = useCase.Run(req.Context(), pkg.FromPegoutConfigurationDTO(request.Configuration))
 		if err != nil {
-			jsonErr := rest.NewErrorResponseWithDetails(UnknownErrorMessage, rest.DetailsFromError(err), false)
-			rest.JsonErrorResponse(w, http.StatusInternalServerError, jsonErr)
+			if errors.Is(err, usecases.TxBelowMinimumError) || errors.Is(err, usecases.NonPositiveWeiError) {
+				jsonErr := rest.NewErrorResponseWithDetails("Validation error", rest.DetailsFromError(err), true)
+				rest.JsonErrorResponse(w, http.StatusBadRequest, jsonErr)
```

Comment:
I think you're missing the return here and since you removed the else at the end it'll try to write in the response writer twice